chain/jsonrpc/src/lib.rs

```diff
@@ -597,16 +596,7 @@ impl JsonRpcHandler {
                 .await;
                 match tx_status_result.clone() {
                     Ok(result) => {
-                        let ready_to_return = if is_legacy_broadcast_tx_commit {
-                            // `broadcast_tx_commit` does not have an option in TxExecutionStatus enum.
-                            // We need to provide the transaction with all the corresponding receipts,
-                            // but we don't want to wait until all of them will be in the canonical chain.
-                            result.execution_outcome.is_some()
-                        } else {
-                            result.status >= finality
-                        };
-
-                        if ready_to_return {
+                        if result.status >= finality {
```

Comment:
Here I just reverted the revert of `broadcast_tix_commit` 
https://github.com/near/nearcore/commit/50bafce559d8d5e2ab9a956943852c284a58884e